### PR TITLE
refactor: remove share role checks

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Use share permissions instead of roles since roles are a very loose concept and just act as a container for a specific set of share permissions.

This affects 2 main scenarios:

* A `SpaceResource` doesn't have properties like `manager`, `editor` anymore. The same goes for methods like `isManager`. Instead, there is the `members` property that holds information about all members. When checking if the current user is allowed to do certain actions, we need check for specific permissions now. E.g. when sharing, don't check for `isManager` but for `canShare`.
* Shares (in Graph terms: "permissions") will also have permissions (in Graph terms: "permission actions") that need to be checked instead of roles. Hence the `GraphShareRoleIdMap` is being removed.

You may notice a discrepancy in our naming and the Graph naming. IMO we should align... but that's a another story, I guess. Because ideally, I would also rename `Space` to `Drive`. It's just so confusing all the time.

There is one exception where Web still needs to assume and work with roles. That is when listing the managers of a space. Here we imply that if the user can remove space members, they are a space manager.

## Related Issue

- refs https://github.com/owncloud/web/issues/10770

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/web-pkg/src/composables/actions/spaces/useSpaceActionsEditReadmeContent.ts [hunk 1]
```diff
@@ -73,10 +78,6 @@ export const useSpaceActionsEditReadmeContent = () => {
           return false
         }
 
-        if (resources[0].disabled) {
-          return false
-        }
-
```
Comment (JammingBen) [inline packages/web-pkg/src/composables/actions/spaces/useSpaceActionsEditReadmeContent.ts:79]: This is already being checked in `canEditReadme`.

File: packages/web-client/src/helpers/share/functions.ts [hunk 2]
```diff
@@ -151,9 +150,7 @@ export function buildIncomingShareResource({
     sharePermissions,
     outgoing: false,
     canRename: () => driveItem['@client.synchronize'],
-    canDownload: () =>
-      sharePermissions.includes(GraphSharePermission.readBasic) &&
-      !shareRoles.some((shareRole) => shareRole.id === GraphShareRoleIdMap.SecureViewer),
+    canDownload: () => sharePermissions.includes(GraphSharePermission.readContent),
```
Comment (kulmann) [inline packages/web-client/src/helpers/share/functions.ts:153]: So do I get it right that `readContent` permission action is not given with a secure view share? Which permission action does the secure view share have for secure-viewing content?

File: packages/web-app-admin-settings/src/components/Spaces/SideBar/MembersPanel.vue [hunk 3]
```diff
@@ -9,103 +9,76 @@
...
         <h3 class="oc-text-bold oc-text-medium" v-text="$gettext('No members found')" />
       </div>
-      <div
-        v-if="filteredSpaceManagers.length"
-        class="oc-mb-m"
-        data-testid="space-members-role-managers"
-      >
-        <h3 class="oc-text-bold oc-text-medium" v-text="$gettext('Managers')" />
-        <members-role-section :members="filteredSpaceManagers" />
+      <div v-for="(role, i) in availableRoles" :key="i">
```
Comment (kulmann) [inline packages/web-app-admin-settings/src/components/Spaces/SideBar/MembersPanel.vue:12]: The order of roles seems to be somewhat random. Seems to be stable per space, but from one space to another it's not the same. Any idea how to sort them in a meaningful and stable way?

Screenshot 2024-08-23 at 11 49 23
Screenshot 2024-08-23 at 11 49 31

Comment (JammingBen) [reply packages/web-app-admin-settings/src/components/Spaces/SideBar/MembersPanel.vue:12]: Hmm I could at least sort by the amount of permissions a role has. That should then lead to Managers > Editors > Viewers.

Comment (kulmann) [reply packages/web-app-admin-settings/src/components/Spaces/SideBar/MembersPanel.vue:12]: nice idea!

File: packages/web-client/src/helpers/share/functions.ts (hunk 2 above)
Comment (JammingBen) [reply packages/web-client/src/helpers/share/functions.ts:153]: Okay so the permissions of a secure view share are these:

```
* libre.graph/driveItem/path/read
* libre.graph/driveItem/children/read
* libre.graph/driveItem/basic/read
```

**Review by kulmann — APPROVED** [review]
Thank you for this! I can imagine that it has been a tedious chunk of work. 🤗

**Pull request merged**